### DISCUSSION TITLE
Assigning fixed ports for each application in its configuration

### DISCUSSION THREAD

**Discussion by deepakag143** [gravitational/teleport#49315] [Q&A]
have installed teleport connect and have configured teleport file with

```
    - name: "devnet15-1600-rpc-call"
      uri: "http://95.216.xxx.xxx:1600/"  # Base URL Teleport proxies to  
      public_addr: "[devnet15-1600.teleport.moinet.io](http://devnet15-1600.teleport.moinet.io/)"       
    - name: "devnet14-1600-rpc-call"
      uri: "http://65.109.xxx.xxx:1600/"  # Base URL Teleport proxies to  
      public_addr: "[devnet14-1600.teleport.moinet.io](http://devnet14-1600.teleport.moinet.io/)"
    - name: "devnet14-1601-rpc-call"
      uri: "http://65.109.xxx.xxx:1601/"  # Base URL Teleport proxies to  
      public_addr: "[devnet14-1601.teleport.moinet.io](http://devnet14-1601.teleport.moinet.io/)"
 ```
so now if i open devnet14-1600-rpc-call in teleport connect :  i m getting :  App Connection ->  Port 62430 ->  Access the app at: http://localhost:62430/ -> Copy i.e it assigns any random port ...i want to fix that port ( i.e i want to give a port to each such application and want it to be static)

**Comment by zmb3** [comment]
Teleport Connect will remember the port and reuse it on subsequent attempts. Alternatively, `tsh proxy app` allows you to specify a port with the `--port` flag.

If you are on macOS, you might consider making these TCP apps instead, as VNet will automatically set up tunnels on demand without you needing to worry about local ports.
